setuptools/command/bdist_wheel.py

```diff
@@ -232,40 +235,47 @@ class bdist_wheel(Command):
 
     def initialize_options(self) -> None:
         self.bdist_dir: str | None = None
-        self.data_dir = None
+        self.data_dir: str | None = None
         self.plat_name: str | None = None
-        self.plat_tag = None
+        self.plat_tag: str | None = None
         self.format = "zip"
         self.keep_temp = False
         self.dist_dir: str | None = None
-        self.egginfo_dir = None
+        self.egginfo_dir: str | None = None
         self.root_is_pure: bool | None = None
-        self.skip_build = None
+        self.skip_build = False
         self.relative = False
         self.owner = None
         self.group = None
         self.universal: bool = False
-        self.compression: str | int = "deflated"
+        self.compression: int = ZIP_DEFLATED
         self.python_tag: str = python_tag()
         self.build_number: str | None = None
         self.py_limited_api: str | Literal[False] = False
         self.plat_name_supplied = False
 
-    def finalize_options(self):
-        if self.bdist_dir is None:
+    def finalize_options(self) -> None:
+        if not self.bdist_dir:
             bdist_base = self.get_finalized_command("bdist").bdist_base
             self.bdist_dir = os.path.join(bdist_base, "wheel")
 
-        egg_info = self.distribution.get_command_obj("egg_info")
+        egg_info = cast(egg_info_cls, self.distribution.get_command_obj("egg_info"))
         egg_info.ensure_finalized()  # needed for correct `wheel_dist_name`
 
         self.data_dir = self.wheel_dist_name + ".data"
-        self.plat_name_supplied = self.plat_name is not None
+        self.plat_name_supplied = bool(self.plat_name)
 
-        try:
-            self.compression = self.supported_compressions[self.compression]
-        except KeyError:
-            raise ValueError(f"Unsupported compression: {self.compression}") from None
+        # Handle compression not being an int or a supported value
+        if not (
+            isinstance(self.compression, int)
+            and self.compression in self.supported_compressions.values()
+        ):
+            try:
+                self.compression = self.supported_compressions[str(self.compression)]
```

Comment:
I can see how the `str(...)` cast is deceiving the typechecker to make it happy: so it believes that `self.compression` is an integer in runtime... but we know it can actually be a string...

Reply:
I think the idea of using a Descriptor or property would clean this up
https://github.com/pypa/setuptools/pull/4383/files#r1646513416